src/Symfony/Component/Yaml/Inline.php

```diff
@@ -375,52 +364,32 @@ private static function parseQuotedScalar($scalar, &$i)
      *
      * @throws ParseException When malformed inline YAML string is parsed
      */
-    private static function parseSequence($sequence, $flags, &$i = 0, $references = array())
+    private static function parseSequence(StringReader $reader, $flags, $references = array())
     {
         $output = array();
-        $len = strlen($sequence);
-        ++$i;
 
         // [foo, bar, ...]
-        while ($i < $len) {
-            switch ($sequence[$i]) {
-                case '[':
-                    // nested sequence
-                    $output[] = self::parseSequence($sequence, $flags, $i, $references);
-                    break;
-                case '{':
-                    // nested mapping
-                    $output[] = self::parseMapping($sequence, $flags, $i, $references);
-                    break;
-                case ']':
-                    return $output;
-                case ',':
-                case ' ':
-                    break;
-                default:
-                    $isQuoted = in_array($sequence[$i], array('"', "'"));
-                    $value = self::parseScalar($sequence, $flags, array(',', ']'), array('"', "'"), $i, true, $references);
-
-                    // the value can be an array if a reference has been resolved to an array var
-                    if (is_string($value) && !$isQuoted && false !== strpos($value, ': ')) {
-                        // embedded mapping?
-                        try {
-                            $pos = 0;
-                            $value = self::parseMapping('{'.$value.'}', $flags, $pos, $references);
-                        } catch (\InvalidArgumentException $e) {
-                            // no, it's not
-                        }
-                    }
+        while (true) {
+            if ($reader->readString(']')) {
+                return $output;
+            } elseif ($reader->readAny(array(',', ' '))) {
```

Comment:
Should it be stricter? Because it looks like it accepts values such as `[ ,, ]` which are not spec compliant (but are supported currently by sf).
That's the same for mappings but I already made it strict (maybe it should be replaced by deprecation warnings).